cmd/acra-translator/server/server.go

```diff
@@ -249,30 +250,51 @@ func (server *ReaderServer) Start(parentContext context.Context) {
 			}
 		}()
 	}
-	if server.config.incomingConnectionGRPCString != "" {
+	// provide way to register new services and custom server
+	if server.config.IncomingConnectionGRPCString() != "" {
 		go func() {
 			grpcLogger := logger.WithField(ConnectionTypeKey, GRPCConnectionType)
-			logger.WithField("connection_string", server.config.incomingConnectionGRPCString).Infof("Start process gRPC requests")
-			secureSessionListener, err := network.NewSecureSessionListener(server.config.ServerID(), server.config.incomingConnectionGRPCString, server.keystorage)
-			if err != nil {
-				grpcLogger.WithError(err).WithField(logging.FieldKeyEventCode, logging.EventCodeErrorTranslatorCantHandleGRPCConnection).
-					Errorln("Can't create secure session listener")
-				return
+			logger.WithField("connection_string", server.config.IncomingConnectionGRPCString()).Infof("Start process gRPC requests")
+			var listener net.Listener
+			var err error
+			var opts []grpc.ServerOption
+			if server.config.WithTLS() {
+				listener, err = network.Listen(server.config.IncomingConnectionGRPCString())
+				if err != nil {
+					grpcLogger.WithError(err).WithField(logging.FieldKeyEventCode, logging.EventCodeErrorTranslatorCantHandleGRPCConnection).
+						Errorln("Can't create gRPC connection listener")
+					return
+				}
+				//listener = tls.NewListener(listener, server.config.GetTLSConfig())
+				opts = append(opts, grpc.Creds(credentials.NewTLS(server.config.GetTLSConfig())))
+			} else {
+				//listener, err = network.NewSecureSessionListener(server.config.ServerID(), server.config.IncomingConnectionGRPCString(), server.keystorage)
+				//if err != nil {
+				//	grpcLogger.WithError(err).WithField(logging.FieldKeyEventCode, logging.EventCodeErrorTranslatorCantHandleGRPCConnection).
+				//		Errorln("Can't create secure session listener")
+				//	return
+				//}
```

Comment:
I guess we don't need this anymore. (And line 268 too.)

Reply:
yes, forgot to remove